Plugins/Public/getrep/Main.cpp

```diff
@@ -0,0 +1,233 @@
+
+#include <windows.h>
+#include <stdio.h>
+#include <string>
+#include <time.h>
+#include <math.h>
+#include <list>
+#include <map>
+#include <algorithm>
+#include <FLHook.h>
+#include <plugin.h>
+#include <PluginUtilities.h>
+#include "Main.h"
+
+#include "../hookext_plugin/hookext_exports.h"
+
+map<string, uint> factions;
+map<string, float> factionReps;
+
+/// A return code to indicate to FLHook if we want the hook processing to continue.
+PLUGIN_RETURNCODE returncode;
+
+void LoadSettings();
+
+BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+	srand((uint)time(0));
+	// If we're being loaded from the command line while FLHook is running then
+	// set_scCfgFile will not be empty so load the settings as FLHook only
+	// calls load settings on FLHook startup and .rehash.
+	if(fdwReason == DLL_PROCESS_ATTACH)
+	{
+		if (set_scCfgFile.length()>0)
+			LoadSettings();
+	}
+	else if (fdwReason == DLL_PROCESS_DETACH)
+	{
+	}
+	return true;
+}
+
+/// Hook will call this function after calling a plugin function to see if we the
+/// processing to continue
+EXPORT PLUGIN_RETURNCODE Get_PluginReturnCode()
+{
+	return returncode;
+}
+
+bool bPluginEnabled = true;
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Loading Settings
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void LoadSettings()
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	INI_Reader ini;
+
+	string factionpropfile = "..\\data\\initialworld.ini";
+	if (ini.open(factionpropfile.c_str(), false))
+	{
+		while (ini.read_header())
+		{
+			if (ini.is_header("Group"))
+			{
+				uint ids_name;
+				string nickname;
+				while (ini.read_value())
+				{
+					if (ini.is_value("nickname"))
+					{
+						nickname = ini.get_value_string();
+					}
+					else if (ini.is_value("ids_name"))
+					{
+						ids_name = ini.get_value_int(0);
+					}
+
+				}
+				factions[nickname] = ids_name;
+			}
+		}
+		ini.close();
+		ConPrint(L"MyRep: Loaded %u factions\n", factions.size());
+	}
+	HkLoadStringDLLs();
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Functions
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+bool UserCmd_Rep(uint iClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
+{
+	wstring wscCharName = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
+
+	for (map<string, uint>::iterator iter = factions.begin(); iter != factions.end(); iter++)
+	{
+		float fRep = 0.0f;
+		string factionName = iter->first;
+		uint factionExternal = iter->second;
+
+		wstring Faction = HkGetWStringFromIDS(factionExternal);
+		uint playerRep = HkGetRep(wscCharName, stows(factionName), fRep);
```

Comment:
this actually returns an HK_ERROR, maybe we should detect it erroring and avoid updating factionReps